src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectConverterFactory.cs

```diff
@@ -39,11 +40,18 @@ public override bool CanConvert(Type typeToConvert)
             Justification = "The ctor is marked RequiresUnreferencedCode.")]
         public override JsonConverter CreateConverter(Type typeToConvert, JsonSerializerOptions options)
         {
+            ConstructorInfo? constructor;
             JsonConverter converter;
             Type converterType;
 
-            bool useDefaultCtorInUnannotatedStructs = _useDefaultConstructorInUnannotatedStructs && !typeToConvert.IsKeyValuePair();
-            if (!typeToConvert.TryGetDeserializationConstructor(useDefaultCtorInUnannotatedStructs, out ConstructorInfo? constructor))
+            if (typeToConvert.IsKeyValuePair())
+            {
+                // browser-wasm compat -- ensure the linker doesn't trim away constructor parameter names from KVP.
```

Comment:
What is this specific to wasm rather than a more general trimming issue?

Reply:
Nothing specific other than it being the only testing platform where we combine reflection with trimming. I should note that special casing KeyValuePair is kind of arbitrary, since similar problems might arise with other types using deserialization constructors. It just happens we have a lot of tests that depend on KVP being rooted, so it made sense to keep it rather than wholesale disable or refactor the tests.